src/main/java/com/db_access/dao/LetterDaoImpl.java

```diff
@@ -0,0 +1,158 @@
+            try (ResultSet result = preparedStatement.executeQuery()) {
+                while (result.next()) {
+                    letters.add(initLetter(result));
+                }
+                closeConnection(preparedStatement);
+            }
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return letters;
+    }
+
+    @Override
+    public void save(Letter letter) {
+        try {
+            PreparedStatement preparedStatement = connection.getConnection().prepareStatement("INSERT INTO letters(" +
+                    "id, recipient,subject,body,deliveryTime) VALUES (?,?,?,?,?)");
+            insertLetterFields(letter, preparedStatement);
+            preparedStatement.executeQuery();
+            closeConnection(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void updateById(Letter letter) {
+        try {
+            PreparedStatement preparedStatement = connection.getConnection().prepareStatement("UPDATE letters SET " +
+                    "id=?, recipient, subject=?, body=?,deliveryTime=?");
+            insertLetterFields(letter, preparedStatement);
+            preparedStatement.executeQuery();
+            closeConnection(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+
+    @Override
+    public void deleteById(Long id) {
+        try {
+            PreparedStatement preparedStatement = connection.getConnection().prepareStatement("DELETE FROM letters WHERE letters.id=?");
+            preparedStatement.setLong(1, id);
+            preparedStatement.executeUpdate();
+            closeConnection(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+    }
+
+    private Letter initLetter(ResultSet resultSet) throws SQLException {
+        Letter letter = new Letter();
+        letter.setId(resultSet.getLong("id"));
+        letter.setBody(resultSet.getNString("body"));
+        letter.setRecipient(resultSet.getNString("recipient"));
+        letter.setDeliveryTime(resultSet.getTimestamp("deliveryTime").toLocalDateTime());
+        return letter;
+    }
+
+    private void insertLetterFields(Letter letter, PreparedStatement statement) throws SQLException {
+        statement.setLong(1, letter.getId());
+        statement.setString(2, letter.getRecipient());
+        statement.setString(3, letter.getSubject());
+        statement.setString(4, letter.getBody());
+        statement.setTimestamp(5, Timestamp.valueOf(letter.getDeliveryTime()));
+    }
+
+    private void closeConnection(PreparedStatement statement) throws SQLException {
+        try {
+            statement.getConnection().commit();
+            if (!statement.getConnection().isClosed())
```

Comment:
1) Auto commit is enabled by default. You do not need to commit manually.
2) Always use code block for any statement. if (...) {...}
3) Connection is always in opened state.